src/app/modules/products/components/product-card/product-card.component.ts

```diff
@@ -8,10 +9,19 @@ import { Product } from "@shared/models/classes/product.class";
     styleUrls: ["./product-card.component.scss"]
 })
 export class ProductCardComponent implements OnInit {
+    @ViewChild("addToCartButton") addToCartButton!: ElementRef;
     @Input() product!: Product;
 
-    constructor() { }
+    constructor(private router: Router) { }
 
     ngOnInit(): void {
     }
+
+    onClick(event: Event): void {
+        const elementUnderClick = event.target as HTMLElement;
+        const addToCartButtonElement = this.addToCartButton.nativeElement as HTMLElement;
+        if (elementUnderClick == addToCartButtonElement) { return; }
+
+        this.router.navigate(["products", this.product.productType, this.product.slug]);
+    }
```

Comment:
this (although a nice solution) is not needed. A simple routerLink in the html will do

Reply:
As discussed this morning, I (just re-)tried the router link with `stopPropagation`, `stopImmediatePropagation`, and `preventDefault` on the click event for the "Add to cart" button, and they either don't stop the router link from activating, or prevent snipcart from seeing the click. It looks like the alternative is `Snipcart.api.cart.items.add({...})` which feels like it undermines the directive a little since we wouldn't need to bind the product details to the element's attributes anymore as we could just pass the details straight into the snipcart API from the component. Should I leave this for now and we can make a new ticket later to create an "addToCart" method in the snipcart service, or do you have an idea of something else I could try?